helm/polaris/build.gradle.kts

```
@@ -0,0 +1,245 @@
+    group = "verification"
+    description = "Run chart-testing (lint)"
+
+    workingDir = rootProject.projectDir
+
+    val outputFile = helmTestReportsDir.get().file("ct/lint.log")
+
+    commandLine =
+      listOf(
+        "sh",
+        "-c",
+        """
+    mkdir -p ${outputFile.asFile.parent}
+    ct lint --debug --charts ./helm/polaris > ${outputFile.asFile} 2>&1
+  """
+          .trimIndent(),
+      )
+
+    inputs.files(
+      fileTree(projectDir) { include("Chart.yaml", "values.yaml", "ci/*.yaml", "templates/**/*") }
+    )
+
+    outputs.file(outputFile)
+  }
+
+// Task to build Docker images in minikube's Docker environment
+val buildMinikubeImages by
+  tasks.registering(Exec::class) {
+    group = "build"
+    description = "Build Polaris Docker images in minikube's Docker environment"
+
+    workingDir = rootProject.projectDir
+
+    // Output: file containing SHA digest of built image
+    val outputFile = helmTestReportsDir.get().file("minikube-images.sha256")
+
+    commandLine =
+      listOf(
+        "sh",
+        "-c",
+        """
+    # Check if minikube is running
+    if ! minikube status >/dev/null 2>&1; then
+      echo "Minikube is not running. Starting minikube..."
+      minikube start
+    fi
+
+    # Set up docker environment and build images
+    eval $(minikube -p minikube docker-env)
+
+    # Build server image
+    docker build -t apache/polaris:latest \
+      -f runtime/server/src/main/docker/Dockerfile.jvm \
+      runtime/server
+
+    # Capture image digest
+    mkdir -p ${outputFile.asFile.parent}
+    {
+      docker inspect --format='{{.Id}}' apache/polaris:latest
+    } > ${outputFile.asFile}
+  """
+          .trimIndent(),
+      )
+
+    dependsOn(":polaris-server:quarkusBuild")
+
+    inputs.dir(rootProject.file("runtime/server/build/quarkus-app"))
+    inputs.file(rootProject.file("runtime/server/src/main/docker/Dockerfile.jvm"))
+
+    outputs.file(outputFile)
+  }
+
+// Task to run chart-testing install on minikube
+val chartTestingInstall by
+  tasks.registering(Exec::class) {
+    group = "verification"
+    description = "Run chart-testing (install) on minikube"
+
+    workingDir = rootProject.projectDir
+
+    val outputFile = helmTestReportsDir.get().file("ct/install.log")
+
+    commandLine =
+      listOf(
+        "sh",
+        "-c",
+        """
+    mkdir -p ${outputFile.asFile.parent}
+
+    # Create namespace if it doesn't exist
+    kubectl create namespace polaris-ns --dry-run=client -o yaml | kubectl apply -f -
+
+    # Install fixtures
+    kubectl apply --namespace polaris-ns -f helm/polaris/ci/fixtures
+
+    # Run chart-testing install
+    ct install \
+      --namespace polaris-ns \
+      --debug --charts ./helm/polaris > ${outputFile.asFile} 2>&1
+  """
+          .trimIndent(),
+      )
+
+    inputs.files(
+      fileTree(projectDir) { include("Chart.yaml", "values.yaml", "ci/**/*", "templates/**/*") }
+    )
+
+    outputs.file(outputFile)
```

All `Exec` tasks also need
```suggestion
    outputs.cacheIf { true }
```
otherwise only up-to-date checks work, but nothing would be cached. (`Exec` is annotated with `@DisableCachingByDefault`.)

---

The cached outputs are (hopefully) relatively small-ish, compared to the execution time of these tasks.